clouddriver-kubernetes/src/main/groovy/com/netflix/spinnaker/clouddriver/kubernetes/v1/deploy/KubernetesUtil.groovy

```diff
@@ -63,32 +63,43 @@ class KubernetesUtil {
     rc.spec?.template?.spec?.imagePullSecrets?.collect({ it.name })
   }
 
-  static KubernetesImageDescription buildImageDescription(String image) {
-    def sIndex = image.indexOf('/')
-    def result = new KubernetesImageDescription()
-
+  private static String extractRegistry(String image, KubernetesImageDescription description) {
+    def index = image.indexOf('/')
     // No slash means we only provided a repository name & optional tag.
-    if (sIndex < 0) {
-      result.repository = image
-    } else {
-      def sPrefix = image.substring(0, sIndex)
+    if (index >= 0) {
+      def sPrefix = image.substring(0, index)
 
       // Check if the content before the slash is a registry (either localhost, or a URL)
       if (sPrefix.startsWith('localhost') || sPrefix.contains('.')) {
-        result.registry = sPrefix
-
-        image = image.substring(sIndex + 1)
+        description.registry = sPrefix
+        image = image.substring(index + 1)
       }
     }
+    image
+  }
 
-    def cIndex = image.indexOf(':')
-
-    if (cIndex < 0) {
-      result.repository = image
+  private static String extractDigestOrTag(String image, KubernetesImageDescription description) {
+    def digestIndex = image.indexOf('@')
```

Comment:
I think `.lastIndexOf('@')` is safer, as the image url could look like `username:password@example.org/repo/image@sha256:abababa`

Reply:
Ah that makes the below check tricky... let me check the spec

Reply:
Looks like the only occurrence of `@` is in the digest, so you're good to go: https://github.com/docker/distribution/blob/master/reference/regexp.go